### PR TITLE
API endpoints for VPN management

### LINKED ISSUE #177: Backend API: VPN management
- add endpoints to apiary
- implement endpoints in django rest framework

### PR DESCRIPTION
**Pull request by gabriel-v**
closes https://github.com/liquidinvestigations/liquidinvestigations/issues/177

### CODE REVIEW

File: liquidcore/config/serializers.py [hunk 1]
```diff
@@ -117,3 +119,25 @@ class RegistrationSerializer(serializers.Serializer):
     lan = LanSerializer()
     wan = WanSerializer()
     ssh = SshSerializer()
+
+class VPNClientKeySerializer(serializers.ModelSerializer):
+    revoked_by = serializers.ReadOnlyField(source='revoked_by.username')
+    def to_representation(self, instance):
+        """Omit null fields from the model"""
+        # stolen from https://stackoverflow.com/a/45569581
```
Comment (mgax) [inline liquidcore/config/serializers.py]: Do we need this comment? Maybe add it to the commit message instead?

File: liquidcore/config/views.py [hunk 2]
```diff
@@ -231,3 +236,124 @@ def post(self, request, format=None):
...
+@api_view(['PUT'])
+@permission_classes([IsAdminUser])
+def vpn_server_enabled(request):
+    serializer = serializers.IsEnabledSerializer(data=request.data)
+    serializer.is_valid(raise_exception=True)
+    enabled = serializer.validated_data['is_enabled']
+    vpn_setting = Setting.objects.get(name='vpn')
+    vpn = vpn_setting.data
+    vpn['server']['is_enabled'] = enabled
+    vpn_setting.data = vpn
```
Comment (mgax) [inline liquidcore/config/views.py]: I get that you have to trigger `setattr` for `data`. If you don't like the back and forth, try a context manager:

```python
with vpn_setting.change_data() as data:
    data['server']['is_enabled'] = enabled
```

And let the context manager overwrite the `data` attribute when you're done.

File: testsuite/test_config_vpn.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    )
+    api_client = APIClient()
+    assert api_client.login(username=USERNAME, password=PASSWORD)
+    return api_client
+
+OVPN_CONTENT_TYPE = 'application/x-openvpn-profile'
+
+def test_vpn_system_status(client):
+    vpn_status = client.get('/api/vpn/')
+    assert 200 == vpn_status.status_code
```
Comment (mgax) [inline testsuite/test_config_vpn.py]: Mindfuck. It looks like you're checking that the number 200 is correct, and `vpn_status.status_code` is your reference value.

**Review by mgax — CHANGES_REQUESTED** [review]

File: liquidcore/config/views.py (hunk 2 above)
Comment (gabriel-v) [reply liquidcore/config/views.py]: Yeah, that makes sense. But won't lose time on it now.

File: liquidcore/config/serializers.py (hunk 1 above)
Comment (gabriel-v) [reply liquidcore/config/serializers.py]: Good idea.

**Pull request merged**